astropy/wcs/wcsapi/fitswcs.py

```diff
@@ -322,7 +322,17 @@ def pixel_to_world_values(self, *pixel_arrays):
         return world[0] if self.world_n_dim == 1 else tuple(world)
 
     def world_to_pixel_values(self, *world_arrays):
-        pixel = self.all_world2pix(*world_arrays, 0)
+        # avoid circular import
+        from astropy.wcs.wcs import NoConvergence
+        try:
+            pixel = self.all_world2pix(*world_arrays, 0)
+        except NoConvergence as e:
+            warnings.warn(str(e))
+            # would be better to use e.best_solution here but that would
+            # have to be formatted by wcs._array_converter which feels like
+            # unnecessary code duplication
+            pixel = self.all_world2pix(*world_arrays, 0, quiet=True)
```

Comment:
What if instead we modified all_world2pix to take a keyword argument to specify that a warning should be emitted instead of an error? This would simplify the code here and also avoid the duplicate call

Reply:
`NoConvergence` exception contains a wealth of information - see https://docs.astropy.org/en/stable/api/astropy.wcs.NoConvergence.html. In particular, using `e.best_solution` should allow one to avoid second call.

Reply:
@mcara As noted in the comment: the result of `best_solution` has a different format than the result that `all_world2pix` returns. Therefore it would be necessary to find out how `all_world2pix` formats the result and duplicate those lines of code in `world_to_pixel_values`.

Reply:
What is the issue with calling `_array_converter()`? Seems to me it is computationally less expensive than solving inverse equations twice.

Reply:
Some time has passed now since I created the PR, but as far as I remember, the code was to complicated for me to find out if it really is just calling `_array_converter` on `best_solution`. I think there was a lot more glue code going on in between.

It we want to use `best_solution`, someone with more knowledge of the code should have a look at it. 

The solution I used is the most robust solution I could think of that assures that the result always has the same formatting, independent of whether a `NoConvergence` error was thrown or not.